utils/cleanup/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	acornCf "github.com/acorn-io/aws/utils/cdk-runner/pkg/aws/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	stackName := os.Getenv("ACORN_EXTERNAL_ID")
+	event := os.Getenv("ACORN_EVENT")
+
+	if event != "delete" {
+		logrus.Error("unexpected event")
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	defer cancel()
+
+	// try to build the necessary clients
+	cfClient, s3Client, err := buildClients(ctx)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	// get the stack
+	stack, err := acornCf.GetStack(&acornCf.Client{Ctx: ctx, Client: cfClient}, stackName)
+	if err != nil {
+		if strings.Contains(err.Error(), "does not exist") {
+			// no cleanup is necessary
+			return
+		}
+
+		logrus.Fatal(err)
+	}
+
+	// check for deletion protection
+	if stack.DeletionProtection && os.Getenv(acornCf.DeletionProtectionEnvKey) == "true" {
```

Comment:
What if the stack has deletion protection and the user is trying to turn it off? thinking out loud, is this correct?
- The event is delete, so the user is trying to remove.
- deletion protection is turned off at the Acorn level, but has not been turned off on the AWS stack side.
- So we will empty the bucket now. 
- The apply job will run with delete protection off, and update the stack, then delete.

If that's correct, then I think we are good.

Reply:
That seems correct. 👍